torchio/transforms/preprocessing/intensity/projection.py

```diff
@@ -0,0 +1,158 @@
+import torch
+from torchio.data.image import ScalarImage
+from ....data.subject import Subject
+from ...intensity_transform import IntensityTransform
+from typing import Optional
+from math import ceil
+
+
+class Projection(IntensityTransform):
+    """Project intensities along a given axis, possibly with sliding slabs.
+
+    Args:
+        axis: Possible inputs are ``'Left'``, ``'Right'``, ``'Anterior'``,
+                ``'Posterior'``, ``'Inferior'``, ``'Superior'``. Lower-case
+                versions and first letters are also valid, as only the first
+                letter will be used.
+        slab_thickness: Thickness of slab projections. In other words, the
+            number of voxels in the ``axis`` dimension to project across.
+            If ``None``, the projection will be done across the entire span of
+            the ``axis`` dimension (i.e. ``axis`` dimension will be reduced to
+            1).
+        stride: Number of voxels to stride along the ``axis`` dimension between
```

Comment:
When is this useful?

Reply:
Which part are you referring to? The projection across the whole dimension or the stride argument? I assume the former. I think the projection across the entire axis is the default behaviour of something like SimpleITK's `sitk.MaximumProjection()` method. I would have to think about it some more, but I think projection across the whole dimension is probably only useful if you've already cropped down the image, otherwise you are projecting across such a large distance that the resulting image probably doesn't make much sense structurally. I'm definitely open to changing this default.

Reply:
After re-reading, I think maybe you are asking about `stride` parameter. Since projections are summarizing transforms of the original image, if `stride < slab_thickness`, for example, the slices in the new image contain structurally overlapping information from the original image.

Setting `stride = slab_thickness` will ensure the new image image is a summary of the old image without overlapping. Does that make sense? Maybe stride should just always be equal to slab thickness and it shouldn't be a configurable argument, unless we can think of good reasons to support strides < or > slab thickness.

Reply:
FWIW, in [this](https://youtu.be/Qmdl0zckFnw?t=367) they talk about moving their slab less than the slab thickness, but perhaps this is only valuable for visualization tools.

Reply:
Yes, I was referring to the stride. Unfortunately, GitHub renders 4 lines when you click on only one (the last one).

I guess my question is "when would a stride > 1 be useful?". It's nice that the user has the choice, but I wonder if it adds unnecessary complexity to the transform. Do you use this argument in your application?

Reply:
Speed considerations might be one reason. If you have a fairly high-resolution image, like a CT scan with dimensions `(512, 512, 300)` and spacing of `(0.7, 0.7, 1.0)` and want 10mm slab projections (`slab_thickness=10`), then striding only 1 in the axial dimension is probably overkill as it only moves 1mm in physical space. Striding 2 (or more) would halve the number of slab computations with (probably) no meaningful loss in structure representation (of course this is context dependent, my brain goes to chest CTs).

Reply:
Ok, that makes sense. Would it be worth it to modify the affine accordingly? I'd need to do the maths, but I guess the spacing along the specified axis could be modified so the image stays in a similar position in world coordinates.

Reply:
Hmm, yes, I suppose it would make sense to update the affine. Should be straightforward enough, I think. Modify the affine value for the chosen `axis` to be `value * self.slab_thickness`?? Something like that.

Reply:
I might be wrong, but I think multiplying by the stride (not thickness) would be the solution.

Reply:
I think the spacing should change with the stride, not with the slab thickness. The current implementation should touch the origin as well, as shown with the results from your snippet:

![SlicerCapture](https://user-images.githubusercontent.com/12688084/131338426-5a4b99c1-72e1-435c-acb5-460b6fd53349.gif)

As you see, the image seems to "move forward" whereas it should approximately stay in the same physical space.

I might have screwed up the axes with my changes, because the projection seems to have happened along the A-P axis, instead of the I-S.

Reply:
> I might have screwed up the axes with my changes, because the projection seems to have happened along the A-P axis, instead of the I-S.

This is because `slab = tensor.index_select(axis_index, slab_indices)` is not taking into account the channels dimension, I'll commit a fix.

Reply:
about the affine change, 
the spacing is the distance (in mm) between two slices, so I would say
original_slice_spacing * stride
note also that the translation part should also be adapted, given the fact there is an implicit croping (compare to the original images) if full_slabs_only is True (same way as affine is modify wen croping)